### PR TITLE
Updated Engine integration scripts to new ruleset structure

### PR DESCRIPTION
**Pull request by JavierBejMen**
|Related issue|
|---|
|#25070|

- Modified engine-suite integration commands to handle the new ruleset structure.
- Added readme.

### CODE REVIEW

**Review by NahuFigueroa97 — APPROVED** [review]
LGTM!
Just nitpicking comments that we could improve in future iterations.

**Review by juliancnn — CHANGES_REQUESTED** [review]
In clean environemnt (new venv), after install all pkg, including engine-suite, an error appears in `-h` parameter:

```bash
(.venv) ╭─root@202-u22-dev ~/repos/wazuh/src/engine/tools/engine-suite ‹enhancement/25070-engine-update-integration-tools●› 
╰─# pip3 install /root/repos/wazuh/src/engine/tools/engine-suite &&                                                                                                                                               1 ↵
pip3 install /root/repos/wazuh/src/engine/tools/api-communication/ &&
pip3 install /root/repos/wazuh/src/engine/test/integration_tests/it-utils &&
engine-integration -h
Processing /root/repos/wazuh/src/engine/tools/engine-suite
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: requests in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from engine-suite==1.0.0.dev0) (2.32.3)
Requirement already satisfied: pyyaml in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from engine-suite==1.0.0.dev0) (6.0.2)
Requirement already satisfied: graphviz in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from engine-suite==1.0.0.dev0) (0.20.3)
Requirement already satisfied: lxml in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from engine-suite==1.0.0.dev0) (5.2.2)
Requirement already satisfied: charset-normalizer<4,>=2 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from requests->engine-suite==1.0.0.dev0) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from requests->engine-suite==1.0.0.dev0) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from requests->engine-suite==1.0.0.dev0) (2.2.2)
Requirement already satisfied: certifi>=2017.4.17 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from requests->engine-suite==1.0.0.dev0) (2024.7.4)
Building wheels for collected packages: engine-suite
  Building wheel for engine-suite (pyproject.toml) ... done
  Created wheel for engine-suite: filename=engine_suite-1.0.0.dev0-py3-none-any.whl size=48372 sha256=4653cdb24167899bfe546dd75c28361ef727c35c0636be875729a718a2baa414
  Stored in directory: /root/.cache/pip/wheels/74/5f/37/96ad11d9b233852186276a8890ce5167d1c68131762a9a88b6
Successfully built engine-suite
Installing collected packages: engine-suite
  Attempting uninstall: engine-suite
    Found existing installation: engine-suite 1.0.0.dev0
    Uninstalling engine-suite-1.0.0.dev0:
      Successfully uninstalled engine-suite-1.0.0.dev0
Successfully installed engine-suite-1.0.0.dev0
Processing /root/repos/wazuh/src/engine/tools/api-communication
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: protobuf==4.21 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from api-communication==1.0.0.dev0) (4.21.0)
Building wheels for collected packages: api-communication
  Building wheel for api-communication (pyproject.toml) ... done
  Created wheel for api-communication: filename=api_communication-1.0.0.dev0-py3-none-any.whl size=26793 sha256=7bb8d6acd0155a0f766c3afa081edcce96131a4920aec1adcf4f870041de83b1
  Stored in directory: /root/.cache/pip/wheels/78/c0/f9/a9392283458abfa94d3994eceebe9e3bfa89e08a5cbb748f75
Successfully built api-communication
Installing collected packages: api-communication
  Attempting uninstall: api-communication
    Found existing installation: api-communication 1.0.0.dev0
    Uninstalling api-communication-1.0.0.dev0:
      Successfully uninstalled api-communication-1.0.0.dev0
Successfully installed api-communication-1.0.0.dev0
Processing /root/repos/wazuh/src/engine/test/integration_tests/it-utils
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: protobuf==4.21 in /root/repos/wazuh/.venv/lib/python3.10/site-packages (from it-utils==1.0.0.dev0) (4.21.0)
Building wheels for collected packages: it-utils
  Building wheel for it-utils (pyproject.toml) ... done
  Created wheel for it-utils: filename=it_utils-1.0.0.dev0-py3-none-any.whl size=3377 sha256=02e4c2a323e87f2de7ee9716014f98af7b29bc3b2bbddf320348ee0807f007c8
  Stored in directory: /root/.cache/pip/wheels/f2/51/9f/6e91068781a57fab01b26dff3c050559108b67bf29edbdd2c7
Successfully built it-utils
Installing collected packages: it-utils
  Attempting uninstall: it-utils
    Found existing installation: it-utils 1.0.0.dev0
    Uninstalling it-utils-1.0.0.dev0:
      Successfully uninstalled it-utils-1.0.0.dev0
Successfully installed it-utils-1.0.0.dev0
Traceback (most recent call last):
  File "/root/repos/wazuh/.venv/bin/engine-integration", line 5, in <module>
    from engine_integration.__main__ import main
  File "/root/repos/wazuh/.venv/lib/python3.10/site-packages/engine_integration/__main__.py", line 10, in <module>
    from .cmds.delete import configure as delete_configure
  File "/root/repos/wazuh/.venv/lib/python3.10/site-packages/engine_integration/cmds/delete.py", line 3, in <module>
    from .generate_manifest import run as gen_manifest
ModuleNotFoundError: No module named 'engine_integration.cmds.generate_manifest'
```
Details
```bash
(.venv) ╭─root@202-u22-dev ~/repos/wazuh/src/engine/tools/engine-suite ‹enhancement/25070-engine-update-integration-tools●› 
╰─# pip3 list                                                                                                                                                                                                   130 ↵
Package            Version
------------------ ----------
api-communication  1.0.0.dev0
certifi            2024.7.4
charset-normalizer 3.3.2
engine-suite       1.0.0.dev0
evtx               0.8.2
evtx2xml           0.1.0
graphviz           0.20.3
idna               3.7
it-utils           1.0.0.dev0
lxml               5.2.2
pip                24.2
protobuf           4.21.0
PyYAML             6.0.2
requests           2.32.3
urllib3            2.2.2
(.venv) ╭─root@202-u22-dev ~/repos/wazuh/src/engine/tools/engine-suite ‹enhancement/25070-engine-update-integration-tools●› 
╰─# python --version
Python 3.10.12
```

File: src/engine/tools/engine-suite/src/engine_integration/cmds/add.py [hunk 1]
```diff
@@ -66,34 +54,31 @@ def run(args, resource_handler: rs.ResourceHandler):
...
+    for type_name in manifest.keys():
+        if type_name not in types:
+            continue
+        path = ruleset_path / type_name
+        if not path.exists():
+            print(f'Error: {type_name} directory does not exist')
+            return -1
+        added_kvdbs_paths = []
+        for entry in path.rglob('*.yml'):
+            if entry.is_file():
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply src/engine/tools/engine-suite/src/engine_integration/cmds/add.py:68]: Done

File: src/engine/tools/engine-suite/src/engine_integration/cmds/update.py [hunk 2]
```diff
@@ -99,33 +66,33 @@ def run(args, resource_handler: rs.ResourceHandler):
...
+    for type_name in manifest.keys():
+        if type_name not in types:
+            continue
+        path = ruleset_path / type_name
+        if not path.exists():
+            print(f'Error: {type_name} directory does not exist')
+            return -1
+
+        for entry in path.rglob('*.yml'):
+            if entry.is_file():
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply src/engine/tools/engine-suite/src/engine_integration/cmds/update.py:79]: done

File: src/engine/tools/engine-suite/src/engine_integration/cmds/add.py [hunk 3]
```diff
@@ -66,34 +54,31 @@ def run(args, resource_handler: rs.ResourceHandler):
...
-                                api_socket, name.split('/')[0], name, original, namespace)
-                            executor.add(task)
+    # Create tasks to add decoders, rules, outputs, filters and kvdbs
+    types = ['decoders', 'rules', 'outputs', 'filters']
+    for type_name in manifest.keys():
+        if type_name not in types:
+            continue
+        path = ruleset_path / type_name
+        if not path.exists():
+            print(f'Error: {type_name} directory does not exist')
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply src/engine/tools/engine-suite/src/engine_integration/cmds/add.py:64]: Done

File: src/engine/tools/engine-suite/src/engine_integration/cmds/generate_doc.py [hunk 4]
```diff
@@ -28,66 +35,39 @@ def run(_, resource_handler: rs.ResourceHandler):
...
-        'decoders', get_asset_doc_fn(resource_handler, decoders_doc))
-    readme_str += f'## Decoders\n\n'
-    readme_str += f'| Name | Description |\n'
-    readme_str += f'|---|---|\n'
-    if decoders_doc:
+    for file in (ruleset_path / 'decoders').rglob('*.yml'):
+        get_asset_doc_fn(resource_handler, decoders_doc)(file.as_posix())
+
+    if len(decoders_doc) > 0:
+        readme_str += f'## Decoders\n\n'
```
[the remark this answers is not in the document]
Comment (JavierBejMen) [reply src/engine/tools/engine-suite/src/engine_integration/cmds/generate_doc.py:45]: left it as it is

**Review by juliancnn — APPROVED** [review]
LGTM!

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**